File: src/main/java/mediathek/gui/actions/EditDownloadAction.java
package mediathek.gui.actions;

import mediathek.gui.tabs.tab_downloads.GuiDownloads;
import mediathek.tool.SVGIconUtilities;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class EditDownloadAction extends AbstractAction {
    private final GuiDownloads guiDownloads;

    public EditDownloadAction(GuiDownloads guiDownloads) {
        this.guiDownloads = guiDownloads;
        putValue(Action.NAME, "Download ändern");
        putValue(Action.SMALL_ICON, SVGIconUtilities.createSVGIcon("icons/fontawesome/pen-to-square.svg"));
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        guiDownloads.editDownload();
    }
}

File: src/main/kotlin/mediathek/gui/actions/EditDownloadAction.kt
package mediathek.gui.actions

import mediathek.gui.tabs.tab_downloads.GuiDownloads
import mediathek.tool.SVGIconUtilities
import java.awt.event.ActionEvent
import javax.swing.AbstractAction

class EditDownloadAction(private val guiDownloads: GuiDownloads) : AbstractAction() {
    init {
        putValue(NAME, "Download ändern...")
        putValue(SMALL_ICON, SVGIconUtilities.createSVGIcon("icons/fontawesome/pen-to-square.svg"))
    }

    override fun actionPerformed(e: ActionEvent?) {
        guiDownloads.editDownload()
    }
}
